relay/relay_test.go

```diff
@@ -562,143 +581,3 @@ func (t *testRelaySuite) verifyMetadata(c *C, r *Relay, uuidExpected string,
 	c.Assert(err, IsNil)
 	c.Assert(UUIDs, DeepEquals, uuidsExpected)
 }
-
-func (t *testRelaySuite) TestProcess(c *C) {
-	var (
-		dbCfg    = getDBConfigForTest()
-		relayCfg = &Config{
-			EnableGTID: false, // position mode, so auto-positioning can work
-			Flavor:     gmysql.MySQLFlavor,
-			RelayDir:   c.MkDir(),
-			ServerID:   12321,
-			From: config.DBConfig{
-				Host:     dbCfg.Host,
-				Port:     dbCfg.Port,
-				User:     dbCfg.User,
-				Password: dbCfg.Password,
-			},
-			ReaderRetry: retry.ReaderRetryConfig{
-				BackoffRollback: 200 * time.Millisecond,
-				BackoffMax:      1 * time.Second,
-				BackoffMin:      1 * time.Millisecond,
-				BackoffJitter:   true,
-				BackoffFactor:   2,
-			},
-		}
-		r = NewRelay(relayCfg).(*Relay)
-	)
-	ctx, cancel := context.WithCancel(context.Background())
-	var wg sync.WaitGroup
-	err := r.Init(ctx)
-	c.Assert(err, IsNil)
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		err2 := r.process(ctx)
-		if !utils.IsErrBinlogPurged(err2) {
-			// we can tolerate `ERROR 1236` caused by `RESET MASTER` in other test cases.
-			c.Assert(err2, IsNil)
-		}
-	}()
-
-	time.Sleep(1 * time.Second) // waiting for get events from upstream
-
-	// kill the binlog dump connection
-	ctx2, cancel2 := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel2()
-	var connID uint32
-	db := r.db.DB
-	c.Assert(utils.WaitSomething(30, 100*time.Millisecond, func() bool {
-		connID, err = getBinlogDumpConnID(ctx2, db)
-		return err == nil
-	}), IsTrue)
-	_, err = db.ExecContext(ctx2, fmt.Sprintf(`KILL %d`, connID))
```

Comment:
not sure this is tested in integration test

Reply:
add a it test in https://github.com/pingcap/dm/pull/2100/commits/42e0dde73c92b0986a7bfa22c3d8031bf1a2fe55